pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -690,38 +688,41 @@ func setSafeDefaults(spec *PerconaXtraDBClusterSpec, log logr.Logger) {
 // setVersion sets the API version of a PXC resource.
 // The new (semver-matching) version is determined either by the CR's API version or an API version specified via the CR's fields.
 // If the CR's API version is an empty string and last-applied-configuration from k8s is empty, it returns current operator version.
-func (cr *PerconaXtraDBCluster) setVersion() (bool, error) {
+func (cr *PerconaXtraDBCluster) setVersion() bool {
 	if len(cr.Spec.CRVersion) > 0 {
-		return false, nil
+		return false
 	}
+
 	apiVersion := version.Version
+
 	if lastCR, ok := cr.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; ok {
 		var newCR PerconaXtraDBCluster
 		err := json.Unmarshal([]byte(lastCR), &newCR)
 		if err != nil {
-			return false, errors.Wrap(err, "unmarshal cr")
+			log.Printf("failed to unmarshal cr: %v", err)
 		}
+
 		if len(newCR.APIVersion) > 0 {
```

Comment:
we should not work with newCR if Unmarshal was unsuccessful 
```suggestion
		} else if len(newCR.APIVersion) > 0 {
```